packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponentProps.java

```diff
@@ -49,8 +52,12 @@ public final class NodeComponentProps implements IProps {
 
     private String parentElementId;
 
+    private Optional<ViewModifier> parentElementState;
```

Comment:
Why is `Optional#empty()` an appropriate value for an enumeration which has a `normal` state? This shouldn't be nullable. It can only create problems.

Reply:
Because nodes may not have a parent and I think it is a semantic error to say that `parentElementState` is `ViewModifier.Normal` if the parent doesn't exist. I can change that if you want.